### PR TITLE
Complete the basic functions of the search bar

### PR DESCRIPTION
**Pull request by tyreseluo**
More content may be needed, only simple room list based filtering has been implemented so far, removal of icon and other content is under construction.

### CODE REVIEW

File: src/sliding_sync.rs [hunk 1]
```diff
@@ -1042,7 +1044,22 @@ fn handle_ignore_user_list_subscriber(client: Client) {
...
+                let left_rooms = client.left_rooms();
+
+                // The all rooms, it includes both joined, invited and left rooms.
+                // Maybe first we need left_rooms to cache, after that we can use it.
+                let all_rooms: Vec<Room> =
+                    joined_rooms
+                    .into_iter()
+                    .chain(invited_rooms)
+                    .chain(left_rooms)
+                    .collect();
```
Comment (kevinaboos) [inline src/sliding_sync.rs:1060]: you don't need to `.collect()` this into a vector if you're just going to iterate over it. You can just iterate over the iterator produced by `.chain()`.

**Review by kevinaboos — CHANGES_REQUESTED** [review]
Hi @tyreseluo, thanks so much for submitting this PR!

I left some very detailed comments, but the summary of my review is that we should not store the list of rooms that match the search query term by duplicating the actual `RoomPreviewEntry`s.

**Comment by tyreseluo** [comment]
Hi @kevinaboos, thank you very much for your comments, I will keep changing and exporting and keep improving myself. Until I am able to complete and think about this project independently. This question will be back to you here within the week. Thanks for your hard work.

**Comment by kevinaboos** [comment]
oh, you don't have to close the issue! I mean, you can close it if you want, or if you prefer to open a new issue that's fine too. But it just needs some moderate restructuring of the way that you store the rooms that match the search terms.

**Comment by tyreseluo** [comment]
oh no! sorry, i made a mistake 😅.  I discard the changes, when i pull the latest code, I don't want to close it. I'd like to keep working on it. i will keep solve this issue!

**Comment by kevinaboos** [comment]
No worries! I'll re-open it.

Let me know when you're ready for another review. Also, kindly avoid force pushing, as it removes my review comments.

**Comment by kevinaboos** [comment]
Re-opening

**Pull request closed without merging**